sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/storage/GcsResourceId.java

```diff
@@ -91,6 +91,17 @@ public String getScheme() {
     return GcsFileSystemRegistrar.GCS_SCHEME;
   }
 
+  @Override
+  public String getFilename() {
+    if (gcsPath.getNameCount() < 1) {
+      return null;
+    } else if (gcsPath.getNameCount() == 1) {
+      return gcsPath.getBucket();
+    } else {
+      return gcsPath.getFileName().toString();
+    }
```

Comment:
On deeper thought, I think `bucket` should be treated like `/` and omitted here. Quick poll around the office -- everyone agreed.

Also, build didn't pass?